src/Cardano/CoinSelection.hs

```diff
@@ -104,23 +107,30 @@ addCoin total c = total + (fromIntegral (getCoin c))
 
 data ErrCoinSelection e
     = ErrNotEnoughMoney Word64 Word64
-    -- ^ UTxO exhausted during input selection
-    -- We record the balance of the UTxO as well as the size of the payment we
+    -- ^ The UTxO was exhausted during input selection.
+    --
+    -- Records the balance of the UTxO, as well as the size of the payment we
     -- tried to make.
-    | ErrUtxoNotEnoughFragmented Word64 Word64
-    -- ^ UTxO is not enough fragmented for the number of transaction outputs
-    -- We record the number of UTxO entries as well as the number of the
-    -- outputs of the transaction.
+    --
+    | ErrUtxoNotFragmentedEnough Word64 Word64
+    -- ^ The UTxO was not fragmented enough to support the required number of
+    -- transaction outputs.
+    --
+    -- Records the number of UTxO entries, as well as the number of the
+    -- transaction outputs.
+    --
     | ErrMaximumInputsReached Word64
-    -- ^ When trying to construct a transaction, the max number of allowed
-    -- inputs was reached.
+    -- ^ The maximum number of allowed inputs was not enough to cover the total
+    -- payment amount.
+    --
     | ErrInputsDepleted
-    -- ^ When trying to construct a transaction, the available inputs are
-    -- depleted even when UTxO is properly fragmented and with enough funds to
-    -- cover payment
+    -- ^ All available inputs were depleted, even though the UTxO was
+    -- sufficiently fragmented and with enough funds to cover payment.
```

Comment:
We don't need to mention construction of transactions at this stage, as we're only dealing with coin selections.